lib/clientEntityContext.ts

```diff
@@ -44,6 +51,11 @@ export interface ClientEntityContextBase {
    * client entity for receiving a batch of messages.
    */
   batchingReceiver?: BatchingReceiver;
+  /**
+   * @property {Dictionary<MessageSession>} messageSessions A dictionary of the MessageSession
+   * objects associated with this client.
+   */
+  messageSessions: Dictionary<MessageSession>;
```

Comment:
Why not use the built-in [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of `Dictionary`? Same question on all other usages of `Dictionary`?

Reply:
Yup we could. Started using a dictionary/JS Object in EventHubs and then looked at Maps while implementing Event Processor Host. Hence started using Maps in some places, since then. 

Was wondering if there should be a preference for one over another. 

Looks like both have their own use cases.
[This article](https://medium.com/front-end-hacking/es6-map-vs-object-what-and-when-b80621932373) provides good info. This [SO link](https://stackoverflow.com/questions/44321324/javascript-map-much-slower-than-object-for-random-look-ups) also provides good info on Maps being slower than objects for lookups.

Reply:
The SO link is a slightly specific case where the keys are all integers.

In the cases that I have seen so far, especially for _clients in the connection context and the messageSessions here, I would prefer using Map, as it would greatly improve readability of the code.

In most cases, we are iterating over the object and then accessing the element from inside the loop. 

Dictionary:
```
for (const messageSession of Object.keys(this._context.messageSessions)) {
       await this._context.messageSessions[messageSession].close();
}
```



for loop on Map
```
for (var [sessionId, messageSession] of this._context.messageSessions) {
   messageSession.close();
}
```

foreach on Map
```
this._context.messageSessions.forEach(messageSession=> messageSession.close());
```

I am not saying we should make these changes in this PR. Just wanted to get your thoughts on the matter